### ISSUE TITLE
Meta: Alloy migration

### ISSUE THREAD

**Issue by Evalir** [foundry-rs/foundry#5216]
Tracking issue for the Alloy Migration.

### Migration progress per-crate:
- [x] `debugger` https://github.com/foundry-rs/foundry/pull/5854
- [x] `fmt` https://github.com/foundry-rs/foundry/pull/5851
- [x] `cli`
- [x] `common`
- [x] `config` partially migrated
- [x] `cast`
  - [x] primitive types migrated
  - [x] signer/provider types migrated
- [x] `macros`
- [x] `abi`
- [x] `evm`
  - [x] Cheatcodes migrated
  - [x] ABI-related types migrated
  - [x] primitive types migrated https://github.com/foundry-rs/foundry/pull/5768
- [x] `anvil-core`
  - [x] https://github.com/foundry-rs/foundry/issues/6715
- [x] `anvil`
  - [x] txpool/executor logic migrated to alloy
  - [x] RPC types migrated
  - [x] primitive types migrated (partially migrated so far) https://github.com/foundry-rs/foundry/pull/5768
  - [x] RLP related types migrated
  - [x] Tests migrated to alloy
- [x] `utils` 
  - [x] primitive types migrated
- [x] `forge`
  - [x]  primitive types migrated https://github.com/foundry-rs/foundry/pull/5873
  - [x] forge-script migrated
- [x] `test-utils migrated https://github.com/foundry-rs/foundry/pull/5874
- [x] `chisel`

### provider/rpc types migration progress per-crate (only ones with usage of these are mentioned):
- [x] `debugger`
- [x] `fmt`
- [x] `cli`
- [x] `common`
- [x] `config`
- [x] `cast`
- [x] `abi`
- [x] `evm`
- [x] `anvil`
- [x] `forge`

done with https://github.com/foundry-rs/foundry/pull/6219

### Reth-related milestones
- [x] Yank reth revm inspectors into their own repo to remove a dependency on reth: https://github.com/paradigmxyz/reth/issues/5956

### soft/hard Blockers
- [x] `Log` alloy-primitives implementation, needed for a complete `anvil` & `evm` migration. https://github.com/alloy-rs/core/pull/271 (Soft blocker)
- [x] `ethers-solc` needs to move to its own standalone repo. (Soft blocker for now)
- [x] `ethers-etherscan` and other verification/etherscan related utilities need to either move to their own repo or integrate into an existing one. (Soft blocker)

### Followups
- #6104 
- #6102 
- #6098 
- https://github.com/foundry-rs/block-explorers/issues/8

**Comment by DaniPopes** [comment]
> I am interested in the migration for `forge bind` . Would be curious to learn more about what all is needed for it.

@0xJepsen 

Forge bind will be deprecated and removed once this migration is done. Bindings generation is no longer needed or possible with the `abigen!` replacement, [`sol!`](https://docs.rs/alloy-sol-macro/latest/alloy_sol_macro/macro.sol.html), as you can either maintain a Solidity interface or a JSON ABI in tree and point the macro to that directly.

**Comment by zerosnacks** [comment]
Marking as complete as Foundry has fully migrated to Alloy :100: 

Leftover code aided in the transition, like https://github.com/foundry-rs/foundry/issues/7048, will be cleaned up in standalone tickets or whenever we encounter it.
